### ISSUE TITLE
YV12 and IYUV format is reversed

### ISSUE THREAD

**Issue by Wilbert-Zhang** [figgis/yuv-viewer#2]
Hello Figgis, 

Firstly, thanks to this wonderful code, basically, it works find to display the YUV picture. 
Howvever, I think the handling to format YV12 and IYUV is reversed. 

I did a experiment using ffmpeg to create a YUV420P YUV picture,  YUV420P is also refered to IYUV/I420/YU12,  but this YUV picture can be displayed with this code only when using option "YV12",  while it's expected to handle this YUV420P YUV picture with option "IYUV". 

I go through the code and try to modify the code at line 400 like:
//-line 400---before change-----
 memcpy(my_overlay->pixels[1], P.cr_data, P.cr_size);
 memcpy(my_overlay->pixels[2], P.cb_data, P.cb_size);
//-------------after change-----------
 memcpy(my_overlay->pixels[1], P.cb_data, P.cb_size);
 memcpy(my_overlay->pixels[2], P.cr_data, P.cr_size);

In the new test with this change applied,  the test result is consistent with ffmpeg behavior. 
Would you please review the this issue. 
Thanks. 

Best regards,
Wilbert

**Comment by figgis** [comment]
True! I mixed them up but I got so used to this that I continue to use YV12 even though the proper notation is IYUV. Should probably fix it some day but I started using vooya instead which is waay better then this :-)

**Comment by ValZapod** [comment]
Is this like NV12 vs NV21? Android only uses the second while Nvidia and windows mostly the first.

NV12 being 8 + 2 + 2 bits per Y, Cb, Cr.

**Comment by Wilbert-Zhang** [comment]
> Is this like NV12 vs NV21? Android only uses the second while Nvidia and windows mostly the first.
> 
> NV12 being 8 + 2 + 2 bits per Y, Cb, Cr, which means 420.

no, it's a bit different in layout of storage.  you can refer to https://www.fourcc.org/yuv/
yv12 is stored like Y Y Y Y..... V V.... U U ..., while nv12 is stored like Y Y Y Y.... U V U V ....

**Comment by ValZapod** [comment]
That is literally what NV12 vs NV21 is all about. They are same, only Cb, Cr swapped.
